includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -0,0 +1,943 @@
+			foreach ( $this->xpath->query( $link_xpath ) as $link ) {
+				if ( $link instanceof DOMElement && preg_match( '/#(.+)/', $link->getAttribute( 'href' ), $matches ) ) {
+					$link->setAttribute( 'on', sprintf( 'tap:%s.scrollTo(duration=600)', $matches[1] ) );
+				}
+			}
+		}
+	}
+
+	/**
+	 * Force SVG support, replacing no-svg class name with svg class name.
+	 *
+	 * @since 1.0
+	 *
+	 * @link https://github.com/WordPress/wordpress-develop/blob/1af1f65a21a1a697fb5f33027497f9e5ae638453/src/wp-content/themes/twentyseventeen/assets/js/global.js#L211-L213
+	 * @link https://caniuse.com/#feat=svg
+	 */
+	public function force_svg_support() {
+		$this->dom->documentElement->setAttribute(
+			'class',
+			preg_replace(
+				'/(^|\s)no-svg(\s|$)/',
+				' svg ',
+				$this->dom->documentElement->getAttribute( 'class' )
+			)
+		);
+	}
+
+	/**
+	 * Force support for fixed background-attachment.
+	 *
+	 * @since 1.0
+	 *
+	 * @link https://github.com/WordPress/wordpress-develop/blob/1af1f65a21a1a697fb5f33027497f9e5ae638453/src/wp-content/themes/twentyseventeen/assets/js/global.js#L215-L217
+	 * @link https://caniuse.com/#feat=background-attachment
+	 */
+	public function force_fixed_background_support() {
+		$this->dom->documentElement->setAttribute(
+			'class',
+			$this->dom->documentElement->getAttribute( 'class' ) . ' background-fixed'
+		);
+	}
+
+	/**
+	 * Add body class when there is a header video.
+	 *
+	 * @since 1.0
+	 * @link https://github.com/WordPress/wordpress-develop/blob/a26c24226c6b131a0ed22c722a836c100d3ba254/src/wp-content/themes/twentyseventeen/assets/js/global.js#L244-L247
+	 *
+	 * @param array $args Args.
+	 */
+	public static function add_has_header_video_body_class( $args = array() ) {
+		$args = array_merge(
+			array(
+				'class_name' => 'has-header-video',
+			),
+			$args
+		);
+
+		add_filter( 'body_class', function( $body_classes ) use ( $args ) {
+			if ( has_header_video() ) {
+				$body_classes[] = $args['class_name'];
+			}
+			return $body_classes;
+		} );
+	}
+
+	/**
+	 * Add required styles for video and image headers.
+	 *
+	 * This is currently used exclusively for Twenty Seventeen.
+	 *
+	 * @since 1.0
+	 * @link https://github.com/WordPress/wordpress-develop/blob/1af1f65a21a1a697fb5f33027497f9e5ae638453/src/wp-content/themes/twentyseventeen/style.css#L1687
+	 * @link https://github.com/WordPress/wordpress-develop/blob/1af1f65a21a1a697fb5f33027497f9e5ae638453/src/wp-content/themes/twentyseventeen/style.css#L1743
+	 */
+	public static function add_twentyseventeen_masthead_styles() {
+		$args = self::get_theme_config( get_template() );
+
+		/*
+		 * The following is necessary because the styles in the theme apply to img and video,
+		 * and the CSS parser will then convert the selectors to amp-img and amp-video respectively.
+		 * Nevertheless, object-fit does not apply on amp-img and it needs to apply on an actual img.
+		 */
+		add_action( 'wp_enqueue_scripts', function() use ( $args ) {
```

Comment:
Similar to the comment above, how about passing a method, instead of an anonymous function? This might help unit testing, but this isn't a blocker now.